File: chatuikit/src/main/java/com/cometchat/chatuikit/extensions/collaborative/CometChatWebViewActivity.java
package com.cometchat.chatuikit.extensions.collaborative;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.webkit.WebResourceRequest;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.cometchat.chatuikit.R;
import com.cometchat.chatuikit.shared.constants.UIKitConstants;

/**
 * CometChatCollaborativeActivity is a Activity class which is used to load
 * extension such as whiteboard and write board in webView.
 *
 * <p>
 * Created On - 26 November 2020
 */

@SuppressLint("SetJavaScriptEnabled")
public class CometChatWebViewActivity extends AppCompatActivity {
    private static final String TAG = CometChatWebViewActivity.class.getSimpleName();

    private String url = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.cometchat_activity_cometchat_webview);
        WebView webView = findViewById(R.id.web_view);
        ImageView backIcon = findViewById(R.id.iv_back);
        TextView textView = findViewById(R.id.tv_title);
        backIcon.setOnClickListener(v -> finish());
        if (getIntent().hasExtra(UIKitConstants.IntentStrings.TITLE)) {
            textView.setText(getIntent().getStringExtra(UIKitConstants.IntentStrings.TITLE));
        }
        if (getIntent().hasExtra(UIKitConstants.IntentStrings.URL)) {
            url = getIntent().getStringExtra(UIKitConstants.IntentStrings.URL);
        }
        if (url != null) {
            WebSettings webSettings = webView.getSettings();
            webSettings.setJavaScriptEnabled(true);
            webSettings.setSupportZoom(true);
            webView.getSettings().setUseWideViewPort(true);
            webSettings.setCacheMode(WebSettings.LOAD_DEFAULT);
            webView.setWebViewClient(new WebViewClient() {
                @Override
                public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
                    return super.shouldOverrideUrlLoading(view, request);
                }
            });
            webView.loadUrl(url);
        }
    }
}

File: uikit-kotlin/src/main/java/screen/CometChatWebViewActivity.kt
package screen

import android.os.Bundle
import android.webkit.WebView
import androidx.appcompat.app.AppCompatActivity
import com.cometchat.pro.uikit.R
import constant.StringContract

class CometChatWebViewActivity : AppCompatActivity() {

    private lateinit var webView: WebView
    private lateinit var url: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_comet_chat_web_view)
        webView = findViewById(R.id.webview)
        if (intent.hasExtra(StringContract.IntentStrings.URL)) {
            url = intent.getStringExtra(StringContract.IntentStrings.URL)
        }
        val webSettings = webView.settings
        webSettings.javaScriptEnabled = true
        webView.loadUrl(url)
    }
}